File: src/main/webapp/js/service/PageService.js
//服务层
app.service('pageService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../page/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../page/findPage.do?pageNum='+page+'&pageSize='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../page/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../page/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../page/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../page/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../page/search.do?pageNum='+page+"&pageSize="+rows, searchEntity);
	}    	
});

File: src/main/java/com/wyan/service/PageService.java
package com.wyan.service;

import com.github.pagehelper.PageInfo;
import com.wyan.pojo.TbPage;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface PageService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<TbPage> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageInfo findPage(int pageNum, int pageSize);


	/**
	 * 增加
	*/
	public void add(TbPage page);


	/**
	 * 修改
	 */
	public void update(TbPage page);


	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public TbPage findOne(Integer catid);


	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Integer[] catids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageInfo findPage(TbPage page, int pageNum, int pageSize);
	
}
